eventbrite-consents/src/main/scala/com/gu/identity/eventbriteconsents/services/ConsentsService.scala

```diff
@@ -0,0 +1,49 @@
+package com.gu.identity.eventbriteconsents.services
+
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+import com.gu.identity.eventbriteconsents.clients.{EventbriteClient, IdentityClient}
+import com.gu.identity.eventbriteconsents.config.LambdaConfig
+import com.typesafe.scalalogging.LazyLogging
+
+class ConsentsService(config: LambdaConfig, eventbriteClient: EventbriteClient, identitiyClient: IdentityClient) extends LazyLogging {
+
+  def syncConsents(): Unit = {
+    val lastRun = Instant.now().minus(config.syncFrequencyHours, ChronoUnit.HOURS)
+    val emailAddresses = findConsentEmails(lastRun, config.masterclassesToken) ++ findConsentEmails(lastRun, config.eventsToken)
+    logger.info(s"Number of email address to sync ${emailAddresses.size}")
+    for {
+      email <- emailAddresses
+    } yield if (config.isDebug) {
+        logger.info(s"Debug flag is set, not syncing $email")
+    } else {
+      identitiyClient.updateEventConsent(email)
+      logger.info(s"Synced $email")
```

Comment:
If we are logging a successfully synced email here, should we not also log the email when `identitiyClient.updateEventConsent(email)` fails?  In case we need to replay any failed updates to Identity?

Reply:
👍 I will add more logging to make it clear which emails the service received successful responses for, which one failed and which ones were never triggered.